### ISSUE TITLE
The corrected figure looks strange.

### ISSUE THREAD

**Issue by Zhangxy1999** [matteo-rizzo/fc4-pytorch#16]
Hi, I'm a beginer in this field. I have a question, is the corrected image in the visualization displayed correctly? Why is there always a slight redness?  
![image](https://github.com/matteo-rizzo/fc4-pytorch/assets/54545114/37a95a43-6bf2-4694-ad7f-70e667c9bc81)
![image](https://github.com/matteo-rizzo/fc4-pytorch/assets/54545114/15c7bd41-dd8c-4a19-9bdc-ae43e8cc42a7)

Looking forward to your early reply.

**Comment by Zhangxy1999** [comment]
It's not 'EPOCH 0', I only test the some images using the premained model. It seems that the code in 'save_vis' function is not correcte. Because the image has been transformed nonlinearly before color correction.
Besides, the performance is not well when there are lights in the image.

**Comment by soleil-de-pluie** [comment]
Thank the author for releasing the code.
I also found this problem. According to #8 , I guess this is because the input of model has done the gamma correction, but the visualization part has not. 
Thus, you can do`torch.pow(img, 2.2)` before correction if using the pretrained model or trained a new model with no gamma correction.

**Comment by Zhangxy1999** [comment]
Thank you for your kind reply. 
I really appreciate the code you shared, it is very helpful to me.

**Comment by matteo-rizzo** [comment]
Thank you @soleil-de-pluie for replying so fast and addressing the issue ad to both of you for using my implementation of FC4

**Comment by flyingwaterdog** [comment]
thank you all, I think this might be (one of) the reasons why correction images appears a little pink. Perhaps during the training procedure in dataloader, 
**img = hwc_to_chw(linear_to_nonlinear(bgr_to_rgb(normalize(img))))**
might be 
**img = hwc_to_chw((bgr_to_rgb(normalize(img))))**
In the TF version of fc4, imgs are not converted to non-linear except for imshow.
I am trying to retrain this model to see if this change during training would work.
Thank the author for this code and all your kind help.

**Comment by flyingwaterdog** [comment]
I have retrained this project and results seems to be more reasonable~
